play/src/front/Phaser/Game/GameScene.ts

```diff
@@ -1333,7 +1333,8 @@ export class GameScene extends DirtyScene {
 
     private safeParseJSONstring(jsonString: string | undefined, propertyName: string) {
         try {
-            return jsonString ? JSON.parse(jsonString) : {};
+            const json = jsonString ? JSON.parse(jsonString) : {};
+            return z.object({}).passthrough().optional().parse(json);
```

Comment:
Check the type instead ?
```ts
            return !json || typeof json === "object" ? json : undefined;
```

Reply:
It is easier, but if the JSON is not an object, I would not get the error message in the catch block. So throwing the error with Zod and catching it below allows me to have one single error message for 2 use cases.